### PR TITLE
EVG-3156: disable patch building with admin flag

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: model/task/task.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type Task struct {
 	// scheduled - the time the commit is scheduled
 	// start - the time the agent starts the task on the host after spinning it up
 	// finish - the time the task was completed on the remote host
-	CreateTime    time.Time `bson:"create_time" json:"create_time"`
+	CreateTime    time.Time `bson:"crate_time" json:"create_time"`
```
Comment (john-m-liu) [inline model/task/task.go]: crate & barrel?

File: scheduler/task_finder.go [hunk 2]
```diff
@@ -72,6 +74,17 @@ func LegacyFindRunnableTasks(distroID string) ([]task.Task, error) {
 			continue
 		}
 
+		if util.StringSliceContains(evergreen.PatchRequesters, task.Requester) && ref.PatchingDisabled {
```
Comment (john-m-liu) [inline scheduler/task_finder.go]: Can you make this + the logging a single function?

File: scheduler/task_finder_test.go [hunk 3]
```diff
@@ -371,6 +382,27 @@ func TestCompareTaskRunnersWithStaticTasks(t *testing.T) {
...
+				Requester: evergreen.GithubPRRequester,
+				Project:   "patching-disabled",
+			},
+			task.Task{
+				Id:        "runnable",
+				Status:    evergreen.TaskUndispatched,
+				Activated: true,
+				Requester: evergreen.RepotrackerVersionRequester,
+				Project:   "patching-disabled",
+			},
```
Comment (john-m-liu) [inline scheduler/task_finder_test.go]: Where are the asserts that go with these?

Comment (tychoish) [reply scheduler/task_finder_test.go]: The asserts are that all methods provide the same results, validated against each others.

**Pull request merged**